pkg/orchestrator/provider/aws/client.go

```diff
@@ -70,34 +70,47 @@ func (c Client) Kind() models.CloudProvider {
 }
 
 // nolint:cyclop
-func (c *Client) DiscoverAssets(ctx context.Context) ([]models.AssetType, error) {
-	regions, err := c.ListAllRegions(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get regions: %w", err)
-	}
+func (c *Client) DiscoverAssets(ctx context.Context) provider.AssetDiscoverer {
+	output := make(chan models.AssetType)
```

Comment:
I wonder if we want to make this channel buffered with a max size of items. Theoretically, we can have a situation where all the assets are inserted into the channel and the memory is still blown